sirepo/job_driver/__init__.py

```diff
@@ -222,8 +224,14 @@ def _websocket_ready_timeout_cancel(self):
             self._websocket_ready_timeout = None
 
     async def _websocket_ready_timeout_handler(self):
-        pkdlog("{} timeout={}", self, self.cfg.agent_starting_secs)
-        await self._start_free_resources(caller="_websocket_ready_timeout_handler")
+        try:
+            if self._websocket_ready_timeout or self._websocket_ready.is_set():
+                pkdlog("ignore timeout {}, is ready", self)
+                return
+            pkdlog("{} timeout={}", self, self.cfg.agent_starting_secs)
+            await self._start_free_resources(caller="_websocket_ready_timeout_handler")
```

Comment:
`TypeError: object NoneType can't be used in 'await' expression`

_start_free_resources is not async.

Reply:
[e1c059c](https://github.com/radiasoft/sirepo/pull/6681/commits/e1c059c3b216f3c81fab82b5fe49385119bc44c9)